### PR TITLE
feat(conversations): create conversation list with last message

### PR DESCRIPTION
**Pull request by mchenani**
**Summary:**
Added a SQL view to represent the list of client conversations, including the last message in each conversation.
- If the last message is an application message, its timestamp is used.
- Otherwise, the group creation time determines the order.

**Context:**
Previously, all conversations were loaded, and the last message for each conversation was fetched individually, resulting in inefficiencies.

**Solution:**
Introduced a SQL-level View to streamline fetching the conversation list, with each row including the latest message, improving performance and clarity.

### CODE REVIEW

File: xmtp_mls/src/storage/encrypted_store/group_message.rs [hunk 1]
```diff
@@ -290,7 +290,7 @@ pub(crate) mod tests {
     use wasm_bindgen_test::wasm_bindgen_test;
     use xmtp_common::{assert_err, assert_ok, rand_time, rand_vec};
 
-    fn generate_message(
+    pub fn generate_message(
```
Comment (codabrink) [inline xmtp_mls/src/storage/encrypted_store/group_message.rs:293]:
```suggestion
    pub(crate) fn generate_message(
```

File: bindings_ffi/src/mls.rs [hunk 2]
```diff
@@ -1141,6 +1161,12 @@ pub struct FfiConversation {
     inner: MlsGroup<RustXmtpClient>,
 }
 
+#[derive(uniffi::Object)]
+pub struct FfiConversationListItem {
```
Comment (mchenani) [inline bindings_ffi/src/mls.rs:1165]: introduced this struct to avoid changing all the current functions!

File: bindings_ffi/src/mls.rs [hunk 3]
```diff
@@ -965,6 +965,26 @@ impl FfiConversations {
         Ok(convo_list)
     }
 
+    pub async fn list_conversations(
```
Comment (mchenani) [inline bindings_ffi/src/mls.rs:968]: default order is:
`ORDER BY COALESCE(rm.sent_at_ns, g.created_at_ns) DESC`

**Review by insipx — APPROVED** [review]
nice!

**Comment by mchenani** [comment]
closes #1181

**Comment by mchenani** [comment]
resolves #1181

File: bindings_ffi/src/mls.rs [hunk 4]
```diff
@@ -947,6 +947,26 @@ impl FfiConversations {
...
```
Comment (nplasterer) [inline bindings_ffi/src/mls.rs:950]: I'm not sure I understand when this is supposed to be used instead of the `list()` function above?

Comment (mchenani) [reply bindings_ffi/src/mls.rs:950]: What I saw in android, we fetched the conversations and the get the last message one by one and map them together. 
we can replace it with this.

**Pull request merged**